gameobject.h

```diff
@@ -11,18 +12,37 @@ namespace pong
 
 class GameObject
 {
+private:
+    static int sId;
+
+    int mId { sId++ };
+
 protected:
     std::unique_ptr<Mesh>mMesh { nullptr };
+    std::unique_ptr<ColliderBox>mColliderBox { nullptr };
+    glm::vec3 mPosition { glm::vec3(0.0f) };
+    std::string mName { "" };
```

Comment:
I would encourage you to just make a GetName() method as a pure virtual, enabling the subclasses to implement these as they need (for instance wall can just return `"Wall"` without messing with mName.

In general, protected data members are a code smell of too-tight coupling between superclasses and subclasses, and should be avoided where ever possible. You may find it to be a maintenance nightmare later on

Reply:
I guess my issue here is that I needed mName to be changeable between instances of the same gameobject. This is just for use with the ball, so it knows which direction to bounce based on which Wall instance was hit.

Reply:
in that case I would possibly recommend a pure virtual getName/setName at some point in the hierarchy instead of a protected member, but either way gets the job done